### ISSUE TITLE
If the slider only contains one slide, it will generate an error

### ISSUE THREAD

**Issue by abstractic** [markirby/FlexSlider-ManualDirectionControls#1]
If the slider only contains one slide, it will generate an error, please see line 23 and the row:

this._originalFlexsliderWrapupFunction = this._flexslider.wrapup;

**Comment by syncnet** [comment]
If you only have one slide then you shouldn't be using a slider in the first place.  Just place your image and you are done!

**Comment by markirby** [comment]
Yes! Indeed!

Mark

**Comment by abstractic** [comment]
Well, I can't control how many slides my customers choose to use, so in the case where they only use one slide, the slider generates an error. I think it's lame, to not control what happens if you remove all slides except one, so that's it.

**Comment by abstractic** [comment]
For your information. You only need to check that the flexslider not are undefined, before you initialize it. A simple solution could be:

<code>
if (this._flexslider != undefined) {

```
this._originalFlexsliderWrapupFunction = this._flexslider.wrapup;
this._defaults = defaults;
this._name = flexsliderManualDirectionControls;
this.init();
```

}
</code>

**Comment by markirby** [comment]
Thanks, although I think you shouldn't need to use this anymore, since the update to FlexSlider which allows manual controls to be used.

Cheers,

Mark

**Comment by abstractic** [comment]
There are still not any support for custom controls in the new FlexSlider, so your code is still, very useful for those who like to move the controls out from the slider element.

Cheers,
Ricki

**Comment by markirby** [comment]
Oh - interesting, I was mistaken. Glad it was worth the effort then!

**Comment by mgratch** [comment]
Currently you can change the controlNav but not the directionNav.  

@abstractic agreed.

**Comment by CHEWX** [comment]
Yup, another +1 for the code. No idea why you cannot customise directionNav but you can controlNav - very dis-jointed.
